src/active_query.rs

```diff
@@ -203,18 +203,27 @@ impl ActiveQuery {
             .is_empty()
             .not()
             .then(|| Box::new(mem::take(accumulated)));
-        let tracked_struct_ids = mem::take(tracked_struct_ids);
+
+        let (verified_final, extra) = if tracked_struct_ids.is_empty() && cycle_heads.is_empty() {
+            (true, None)
+        } else {
+            let extra = QueryRevisionsExtra {
+                tracked_struct_ids: mem::take(tracked_struct_ids),
+                cycle_heads: mem::take(cycle_heads),
+            };
```

Comment:
Nit: I think it would be nice if `QueryRevisionsExtra` could abstract the fact that it is an `Option<Box<Data>>` internally to keep the code here simpler (`QueryRevisions::new(final, tracked_struct_ids, cycle_heads)`)